File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> keyinput, vector<int> board) {
    
    vector<int> answer;
    
    answer.push_back(0);
    answer.push_back(0);
    
    for (int i=0; i<keyinput.size(); i++) {
        if (keyinput[i] == "left") {
            if ( abs(answer[0] - 1) <= board[0] / 2 ) answer[0] -= 1;
        } else if (keyinput[i] == "right") {
            if ( abs(answer[0] + 1) <= board[0] / 2 ) answer[0] += 1;
        } else if (keyinput[i] == "up") {
            if ( abs(answer[1] + 1) <= board[1] / 2 ) answer[1] += 1;
        } else {
            if ( abs(answer[1] - 1) <= board[1] / 2 ) answer[1] -= 1;
        }
    }
    
    return answer;
}

File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    pos = [0, 0]
    board = [board[0] // 2, board[1] // 2]
    
    for key in keyinput:
        if key == 'left' and abs(pos[0] - 1) <= board[0]:
            pos[0] -= 1
        elif key == 'right' and abs(pos[0] + 1) <= board[0]:
            pos[0] += 1
        elif key == 'up' and abs(pos[1] + 1) <= board[1]:
            pos[1] += 1
        elif key == 'down' and abs(pos[1] - 1) <= board[1]:
            pos[1] -= 1
            
    return pos